services/command-router/src/main/java/org/eclipse/hono/commandrouter/CommandTargetMapper.java

```diff
@@ -0,0 +1,111 @@
+/*******************************************************************************
+ * Copyright (c) 2019, 2021 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+
+package org.eclipse.hono.commandrouter;
+
+import org.eclipse.hono.adapter.client.registry.DeviceRegistrationClient;
+import org.eclipse.hono.client.ClientErrorException;
+import org.eclipse.hono.client.ServiceInvocationException;
+import org.eclipse.hono.commandrouter.impl.CommandTargetMapperImpl;
+import org.eclipse.hono.deviceconnection.infinispan.client.DeviceConnectionInfo;
+import org.eclipse.hono.util.DeviceConnectionConstants;
+
+import io.opentracing.SpanContext;
+import io.opentracing.Tracer;
+import io.vertx.core.Future;
+import io.vertx.core.json.JsonObject;
+
+/**
+ * A component for determining where a incoming command message should be targeted at when processing the message in a
+ * protocol adapter.
+ * <p>
+ * This refers to finding the <em>protocol adapter instance</em> that a device or gateway has connected to in order to
+ * receive a command message. Furthermore, it is determined to which <em>gateway</em> a command is to be mapped to or
+ * whether it will be sent to the target device directly.
+ * <p>
+ * For obtaining both information, the <em>getCommandHandlingAdapterInstances</em> operation of the Device Connection
```

Comment:
isn't that an implementation detail from now on? I was under the impression that we are replacing the Device Connection API with the Command Router API ...

Reply:
Yes, good point. I've changed the javadoc now.